core/keyboard_nav/navigation.js

```diff
@@ -857,70 +902,29 @@ Blockly.navigation.toolboxOnAction_ = function(action) {
 };
 
 /**
- * Enable accessibility mode.
- */
-Blockly.navigation.enableKeyboardAccessibility = function() {
-  if (!Blockly.keyboardAccessibilityMode) {
-    Blockly.keyboardAccessibilityMode = true;
-    Blockly.navigation.focusWorkspace_();
-  }
-};
-
-/**
- * Disable accessibility mode.
- */
-Blockly.navigation.disableKeyboardAccessibility = function() {
-  if (Blockly.keyboardAccessibilityMode) {
-    Blockly.keyboardAccessibilityMode = false;
-    Blockly.getMainWorkspace().getCursor().hide();
-    Blockly.getMainWorkspace().getMarker().hide();
-    if (Blockly.navigation.getFlyoutCursor_()) {
-      Blockly.navigation.getFlyoutCursor_().hide();
-    }
-  }
-};
-
-/**
- * Navigation log handler. If loggingCallback is defined, use it.
- * Otherwise just log to the console.
- * @param {string} msg The message to log.
- * @package
- */
-Blockly.navigation.log_ = function(msg) {
-  if (Blockly.navigation.loggingCallback) {
-    Blockly.navigation.loggingCallback('log', msg);
-  } else {
-    console.log(msg);
-  }
-};
-
-/**
- * Navigation warning handler. If loggingCallback is defined, use it.
- * Otherwise call Blockly.navigation.warn_.
- * @param {string} msg The warning message.
- * @package
+ * Handles hitting the enter key on the workspace.
+ * @private
  */
-Blockly.navigation.warn_ = function(msg) {
-  if (Blockly.navigation.loggingCallback) {
-    Blockly.navigation.loggingCallback('warn', msg);
-  } else {
-    console.warn(msg);
+Blockly.navigation.handleEnterForWS_ = function() {
+  var cursor = Blockly.getMainWorkspace().getCursor();
+  var curNode = cursor.getCurNode();
+  var nodeType = curNode.getType();
+  if (nodeType === Blockly.ASTNode.types.FIELD) {
```

Comment:
if/elseif could be changed to switch statement

Reply:
I actually don't think i can here because of the curNode.isConnection()

Reply:
you could put an if statement inside of case for 'Blockly.ASTNode.types.WORKSPACE' to check for isConnection

Reply:
I think whether it is a connection is separate from a whether it is a workspace. If it is either a workspace or a connection then it should mark at the cursor